bike_rent/models/sale.py

```diff
@@ -34,7 +40,7 @@ class SaleOrderLine(models.Model):
     @api.multi
     def _compute_rent_end_date(self):
         for line in self.filtered(lambda x: x.product_id.type == 'service' and x.product_id.is_bike):
-            if line.state == 'draft' or line.state == 'sent':
-                line.rent_end_date = fields.Datetime.now() + timedelta(days=line.product_id.rent_duration)
-            else:
+            if line.bike_rent_id.rent_stop:
                 line.rent_end_date = line.bike_rent_id.rent_stop
+            else:
+                line.rent_end_date = fields.Datetime.now() + timedelta(days=line.product_id.rent_duration)
```

Comment:
if a sale order was created in a `draft` state a single day before, this would still change the `rent_end_date`.

Reply:
Which is intended, as officially we start the rent once the order is confirmed, and not when the quotation(draft) is first made. So while the order is not confirmed, we represent latest possible return date in our reports, i.e. datetime.now() (since the idea was that we print the report to the customer for signing as confirmation).